feat: Add context menu to ribbon icon

SrGeneroso opened the pull request:
I've added a context menu into the ribbon icon.
![image](https://github.com/user-attachments/assets/117c7dfa-8679-47ae-ba13-45d37d563340)
Now you can push and pull from the same icon.

SrGeneroso commented:
Should I add also reset?
![image](https://github.com/user-attachments/assets/211ba12d-0d42-4f78-8dff-c88657c4f0fe)
I didn't because it would be awful in case of miss click.
Let me know.

mrhillsman commented:
If you can add a confirmation to the reset then I think it would be fine. You might want to add confirmation for all when clicked but at a minimum confirmation for reset is a must. Otherwise I would not add it.

SrGeneroso commented:
Ok, so I added the reset action to the ribbon.
![image](https://github.com/user-attachments/assets/38b5bb5a-da82-48d9-b292-b4ebf4e47154)
I added as well a modal for confirmation
![image](https://github.com/user-attachments/assets/4b46d2fe-5da8-46b2-8e70-cfe974ac03c0)
Might be wise to revisit the modal I created as it started as a modal confirmation that could be customized for the other actions, but I felt that it was only necessary for the Reset action, as the other one already have modals in place. Might be better to hardcode it? Let me know, in any case it seems to work.

SrGeneroso commented:
I think this is it. The modal now lives in the reset helper the same as the others helpers. The message are hardcoded and therefor simplified based on the modals already present.
As a nice addition, I've made the ribbon not appear if there is a syncing in process.
Check it out and let me know. Most likely you would like to change the messages.

SrGeneroso commented:
Finally some touches and styling. I've learnt a lot about obsidian plugins this week! web tech is awesome.
This is the final form of the changes.
![image](https://github.com/user-attachments/assets/6dbfa58e-b8f7-48ee-adf9-2734bf54fe95)

RichardX366 commented:
I'll test this when I get home and then merge since it looks pretty good.

SrGeneroso commented:
> I'll test this when I get home and then merge since it looks pretty good.

Please consider proofreading the text, my grammar could improve.

AzCii commented:
Out of curiosity, why where this never merged?

SrGeneroso commented:
Great!, I hope it is helpful. I was excited to see my plugin was updated as well, until I remembered that I manually edited my files and I'm the only one with this functionality available. I hope it gets its own release.
